airflow/jobs.py

```diff
@@ -538,12 +538,17 @@ def manage_slas(self, dag, session=None):
         Where assuming that the scheduler runs often, so we only check for
         tasks that should have succeeded in the past hour.
         """
+        if not any([ti.sla for ti in dag.tasks]):
+            self.logger.info('Skipping SLA check for {} because no tasks in DAG have SLAs'.format(dag))
+            return
+
         TI = models.TaskInstance
         sq = (
             session
             .query(
                 TI.task_id,
                 func.max(TI.execution_date).label('max_ti'))
+            .with_hint(TI, 'USE INDEX (PRIMARY)', dialect_name='mysql')
```

Comment:
Acceptable for now: issue seems to be more prevalent as primary almost never seems to be used. The query optimiser is really weird here.